machine/accton/accton_as_28t_ops/u-boot/platform-as_28t_ops.patch

```diff
@@ -0,0 +1,2795 @@
++			 * then change it.
++			 */
++			char *env;
++			env = getenv(enetvar);
++			if (!env) {
++				setenv(enetvar, ethaddr);
++			} else {
++				if(strcmp(env, ethaddr))
++					setenv(enetvar, ethaddr);
++			}
++}
++#else
+ 			/* Only initialize environment variables that are blank
+ 			 * (i.e. have not yet been set)
+ 			 */
+ 			if (!getenv(enetvar))
+ 				setenv(enetvar, ethaddr);
++#endif
+ 
+ 			macbase[5]++;
+ 			if (macbase[5] == 0) {
+diff --git a/common/env_common.c b/common/env_common.c
+index d6162a4..9387b67 100644
+--- a/common/env_common.c
++++ b/common/env_common.c
+@@ -31,6 +31,7 @@
+ #include <search.h>
+ #include <errno.h>
+ #include <malloc.h>
++#include <spi_flash.h>
+ 
+ DECLARE_GLOBAL_DATA_PTR;
+ 
+@@ -328,10 +329,35 @@ int env_complete(char *var, int maxv, char *cmdv[], int bufsz, char *buf)
+ */
+ void env_write_default_if_crc_bad(void)
+ {
+-	env_t *env_ptr = (env_t *)CONFIG_ENV_ADDR;
++	int ret;
++	env_t *env_ptr = (env_t *)malloc(sizeof(env_t));
++	memset(env_ptr, '\0', sizeof(env_t));
++
++	struct spi_flash *env_flash = NULL;
++
++	if (!env_flash) {
++		env_flash = spi_flash_probe(CONFIG_ENV_SPI_BUS,
++				CONFIG_ENV_SPI_CS,
++				CONFIG_ENV_SPI_MAX_HZ, CONFIG_ENV_SPI_MODE);
++		if (!env_flash) {
++			set_default_env("!spi_flash_probe() failed");
++			goto free_and_exit;
++		}
++	}
++
++	ret = spi_flash_read(env_flash, CONFIG_ENV_OFFSET, ENV_SIZE, (char *)env_ptr);
++	if (ret) {
++		printf("SPI Read Fail!\n");
++		goto free_and_exit;
++	}
++
+ 	if (crc32(0, env_ptr->data, ENV_SIZE) != env_ptr->crc) {
+ 		puts("Writing default environment\n");
+ 		saveenv();
+ 	}
++
++free_and_exit:
++	free(env_ptr);
++	return;
+ }
+ #endif
+diff --git a/common/image.c b/common/image.c
+index f084d2b..d3ea426 100644
+--- a/common/image.c
++++ b/common/image.c
+@@ -1402,10 +1402,14 @@ int boot_get_fdt(int flag, int argc, char * const argv[],
+ 	const void	*data;
+ 	size_t		size;
+ #endif
++	const char	*select = NULL;
++	int		ok_no_fdt = 0;
+ 
+ 	*of_flat_tree = NULL;
+ 	*of_size = 0;
+ 
++	if (argc > 3)
++		select = argv[3];
+ 	if (argc > 3 || genimg_has_config(images)) {
+ #if defined(CONFIG_FIT)
+ 		if (argc > 3) {
+@@ -1491,7 +1495,7 @@ int boot_get_fdt(int flag, int argc, char * const argv[],
+ 					fdt_addr);
+ 			fdt_hdr = image_get_fdt(fdt_addr);
+ 			if (!fdt_hdr)
+-				goto error;
++				goto no_fdt;
+ 
+ 			/*
+ 			 * move image data to the load address,
+@@ -1651,7 +1655,7 @@ int boot_get_fdt(int flag, int argc, char * const argv[],
+ 		default:
+ 			puts("ERROR: Did not find a cmdline Flattened Device "
+ 				"Tree\n");
+-			goto error;
++			goto no_fdt;
+ 		}
+ 
+ 		printf("   Booting using the fdt blob at 0x%p\n", fdt_blob);
+@@ -1678,8 +1682,8 @@ int boot_get_fdt(int flag, int argc, char * const argv[],
+ 			printf("   Booting using the fdt at 0x%p\n", fdt_blob);
+ 
+ 			if (fdt_check_header(fdt_blob) != 0) {
+-				fdt_error("image is not a fdt");
+-				goto error;
```

Comment:
Looks like changes to support no_fdt seems to be common across all iproc platforms. 
Making this patch as common iproc patch will be good future cleanup